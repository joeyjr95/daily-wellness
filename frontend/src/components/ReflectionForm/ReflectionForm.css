.mental_rating {
    float: left;
    height: 46px;
    padding: 0 10px;
}
.mental_rating:not(:checked) > input {
    position:absolute;
    top:-9999px;
}
.mental_rating:not(:checked) > label {
    float:right;
    width:1em;
    overflow:hidden;
    white-space:nowrap;
    cursor:pointer;
    font-size:30px;
    color:#ccc;
}
.mental_rating:not(:checked) > label:before {
    content: 'â˜… ';
}
.mental_rating > input:checked ~ label {
    color: #ffc700;    
}
.mental_rating:not(:checked) > label:hover,
.mental_rating:not(:checked) > label:hover ~ label {
    color: #deb217;  
}
.mental_rating > input:checked + label:hover,
.mental_rating > input:checked + label:hover ~ label,
.mental_rating > input:checked ~ label:hover,
.mental_rating > input:checked ~ label:hover ~ label,
.mental_rating > label:hover ~ input:checked ~ label {
    color: #c59b08;
}