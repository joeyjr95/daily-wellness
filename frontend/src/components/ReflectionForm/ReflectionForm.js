import React, { Component } from 'react'
import WellnessContext from '../../contexts/WellnessContext'
import WellnessApiService from '../../services/wellness-api-service'
import { Button } from '../Utils/Utils'

export default class ReflectionForm extends Component {
    static defaultProps = {
        onFormSuccess: () => {}
      }

    static contextType = WellnessContext

    handleSubmit = e => {
        e.preventDefault()
        const reflection = {
            user_id: this.context.user,
            physical_rating: Number(e.target['physical_rating'].value),
            physical_content: e.target['physical_content'].value,
            mental_rating: Number(e.target['mental_rating'].value),
            mental_content: e.target['mental_content'].value,
        }
        WellnessApiService.postReflection(reflection)
            .then(this.context.addReflection,
                  this.props.onFormSuccess())
            .catch(this.context.setError)
    }

    render() {

        return(
        <>
          <form onSubmit ={this.handleSubmit}>
                <div>
                     <label htmlFor="mental_rating">Rate your mental health today</label>
                    <select name='mental_rating' id='mental_rating'>
                        <option value="1">1</option>
                        <option value="2">2</option>
                        <option value="3">3</option>
                        <option value="4">4</option>
                        <option value="5">5</option>
                    </select>
                </div>
            <div>
            <textarea name="mental_content" id="mental_content"rows="10" cols="30">explain your mental rating</textarea>
           </div>
           <div>
                <label htmlFor="physical_rating">Rate your physical health today</label>
                <select name='physical_rating' id='physical_rating'>
                    <option value="1">1</option>
                    <option value="2">2</option>
                    <option value="3">3</option>
                    <option value="4">4</option>
                    <option value="5">5</option>
                </select>
            </div>
           <div> 
           <textarea name="physical_content" id="physical_content" rows="10" cols="30">explain your physical rating</textarea> 
           </div>
           <Button type='submit'>
          Submit Reflection
        </Button>
         </form>

        </>
        )
    }

}