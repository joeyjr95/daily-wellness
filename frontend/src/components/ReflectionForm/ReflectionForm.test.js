import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReflectionForm from './ReflectionForm';

describe(`ReflectionForm component`, () => {
  it('renders a ReflectionForm by default', () => {
    const wrapper = shallow(<ReflectionForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})