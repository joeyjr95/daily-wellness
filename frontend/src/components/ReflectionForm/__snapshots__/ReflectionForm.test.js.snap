// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReflectionForm component renders a ReflectionForm by default 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="mental_rating"
      >
        Rate your mental health today
      </label>
      <select
        id="mental_rating"
        name="mental_rating"
      >
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
        <option
          value="3"
        >
          3
        </option>
        <option
          value="4"
        >
          4
        </option>
        <option
          value="5"
        >
          5
        </option>
      </select>
    </div>
    <div>
      <textarea
        cols="30"
        id="mental_content"
        name="mental_content"
        placeholder="leave a description here!"
        rows="10"
      />
    </div>
    <div>
      <label
        htmlFor="physical_rating"
      >
        Rate your physical health today
      </label>
      <select
        id="physical_rating"
        name="physical_rating"
      >
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
        <option
          value="3"
        >
          3
        </option>
        <option
          value="4"
        >
          4
        </option>
        <option
          value="5"
        >
          5
        </option>
      </select>
    </div>
    <div>
      <textarea
        cols="30"
        id="physical_content"
        name="physical_content"
        placeholder="leave a description here!"
        rows="10"
      />
    </div>
    <button
      type="submit"
    >
      Submit Reflection
    </button>
  </form>
</Fragment>
`;
